spec/features/visitors/email_confirmation_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+feature 'Email confirmation during sign up' do
+  scenario 'confirms valid email and sets valid password' do
+    email = 'test@example.com'
+    sign_up_with(email)
+    raw_confirmation_token, = Devise.token_generator.generate(User, :confirmation_token)
+    User.find_by(email: email).update(
+      confirmation_token: raw_confirmation_token, confirmation_sent_at: Time.current
+    )
+
+    visit "/users/confirmation?confirmation_token=#{raw_confirmation_token}"
+
```

Comment:
The way the confirmation token is generated and accessed here is setup to me because that's what Devise does in the background, and we're not testing Devise here. The user does not perform any of these tasks.

If we wanted a true integration test, we would not cheat by invoking Devise. We would have the user open up their email and click on the appropriate button or link.

If we don't already have a test that opens up the email and clicks on the confirmation link, then this is probably the place to do it. If we do, then we already have that core feature covered, and we can use helpers in this and other tests. What do you think?